### PR TITLE
[hariom] developed modal and response page for workflow attendance

### PR DESCRIPTION
**Pull request by hariomsinha-eGov**

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/config/configCheckModal.js [hunk 1]
```diff
@@ -1,105 +1,106 @@
-import { Dropdown,Loader } from '@egovernments/digit-ui-react-components';
-import React,{useState} from 'react'
+import { Dropdown, Loader } from "@egovernments/digit-ui-react-components";
+import React, { useState } from "react";
 
 const configCheckModal = ({
```
Comment (nipunarora-eGov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/config/configCheckModal.js:4]: @hariomsinha-eGov discard all changes in these 3 works modals files

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/config/configRejectModal.js [hunk 2]
```diff
@@ -14,60 +14,63 @@ const configRejectModal = ({
 
     let checkConditions = true
     if (action.isTerminateState) checkConditions = false;
-
+    const fieldLabelStyle = {
+        "display" : "grid",
```
Comment (nipunarora-eGov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/config/configRejectModal.js:18]: @hariomsinha-eGov discard all changes here

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/config/configApproveModal.js [hunk 3]
```diff
@@ -1,53 +1,49 @@
...
-}) => {
-    if(action?.action === 'ADMINSANCTION'){
+const configApproveModal = ({ t, action }) => {
+  if (action?.action === "ADMINSANCTION") {
     return {
-        label: {
-            heading: `WORKS_APPROVE_ESTIMATE`,
-            submit: `WORKS_APPROVE_ESTIMATE`,
-            cancel: "CS_COMMON_CANCEL",
-        },
```
Comment (nipunarora-eGov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/config/configApproveModal.js:14]: @hariomsinha-eGov discard all changes in this file

**Review by nipunarora-eGov — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+   const [department, setDepartment] = useState([]);
+   const [selectedDept,setSelectedDept] = useState({})
+
+   const [rejectionReason, setRejectionReason] = useState([]);
+   const [selectedReason,setSelectedReason] = useState([])
+
+   const [designation, setDesignation] = useState([]);
+   const [selectedDesignation,setSelectedDesignation] = useState({})
+
+  const rejectReasons = [
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js:41]: can u push into mdms ?

Comment (jagankumar-egov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js:41]: and integrate

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+      {
+        "name": "Designation"
+      },
+      {
+        "name": "Department"
+      }
+    ]
+  );
+
+  mdmsData?.["common-masters"]?.Designation?.map(designation => {
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js:75]: for formatting any hook data send a function as default which will give u formatted response
ref : frontend/micro-ui/web/micro-ui-internals/packages/libraries/src/hooks/useCustomMDMS.js

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js (hunk 4 above)
Comment (hariomsinha-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js]: I have pushed the new changes.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js (hunk 5 above)
Comment (hariomsinha-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js]: done

**Review by jagankumar-egov — APPROVED** [review]

**Pull request merged**